gluecodium/src/main/resources/templates/ffi/FfiBlobHandleHeader.mustache

```diff
@@ -30,14 +30,11 @@
 extern "C" {
 #endif
 
-_GLUECODIUM_FFI_EXPORT FfiOpaqueHandle blob_create_handle();
+_GLUECODIUM_FFI_EXPORT FfiOpaqueHandle blob_create_handle(uint64_t length);
 _GLUECODIUM_FFI_EXPORT void blob_release_handle(FfiOpaqueHandle handle);
-_GLUECODIUM_FFI_EXPORT void blob_insert(FfiOpaqueHandle handle, uint8_t value);
-_GLUECODIUM_FFI_EXPORT FfiOpaqueHandle blob_iterator(FfiOpaqueHandle handle);
-_GLUECODIUM_FFI_EXPORT void blob_iterator_release_handle(FfiOpaqueHandle iterator_handle);
-_GLUECODIUM_FFI_EXPORT bool blob_iterator_is_valid(FfiOpaqueHandle handle, FfiOpaqueHandle iterator_handle);
-_GLUECODIUM_FFI_EXPORT void blob_iterator_increment(FfiOpaqueHandle iterator_handle);
-_GLUECODIUM_FFI_EXPORT uint8_t blob_iterator_get(FfiOpaqueHandle iterator_handle);
+_GLUECODIUM_FFI_EXPORT uint64_t blob_get_length(FfiOpaqueHandle iterator_handle);
+_GLUECODIUM_FFI_EXPORT uint8_t blob_get_at(FfiOpaqueHandle iterator_handle, uint64_t index);
```

Comment:
Already better, but I was wondering whether or not you could pass a pointer to the underlying ByteBuffer? It seems there is no easy way to assign Int8pointer content to a ByteBuffer in Dart, but you could get the list length from FFI, create UInt8List in Dart and then pass the underlying ByteBuffer to FFI to copy the values into it via std::copy or memcpy.

Reply:
I'm not sure if this is even possible. I created a spike #115 to investigate this later.